sormas-backend/src/main/java/de/symeda/sormas/backend/externaljournal/ExternalJournalService.java

```diff
@@ -312,19 +312,20 @@ public PatientDiaryRegisterResult registerPatientDiaryPerson(PersonDto person) {
 	private Invocation.Builder getExternalDataPersonInvocationBuilder(String personUuid) {
 		String externalDataUrl = configFacade.getPatientDiaryConfig().getProbandsUrl() + "/external-data/" + personUuid;
 		Client client = ClientBuilder.newClient();
-		return client.target(externalDataUrl)
-				.request(MediaType.APPLICATION_JSON)
-				.header("x-access-token", getPatientDiaryAuthToken());
+		return client.target(externalDataUrl).request(MediaType.APPLICATION_JSON).header("x-access-token", getPatientDiaryAuthToken());
 	}
 
 	/**
-	 * Check whether a person has valid data in order to be registered in the patient diary
 	 * 
 	 * @param person
 	 *            the person to validate
+	 * @param calledForTest
+	 *            defines if method makes a call to a patient diary to check for availability of phone and mail addresses.
+	 *            In tests, this call is not made.
+	 *            This boolean should be false unless it's called from tests.
 	 * @return the result of the validation
 	 */
-	public PatientDiaryPersonValidation validatePatientDiaryPerson(PersonDto person) {
+	public PatientDiaryPersonValidation validatePatientDiaryPerson(PersonDto person, boolean calledForTest) {
```

Comment:
I don't think we should add different functionality specifically for tests, since the tested method should work exactly as if it would be called outside a test. If the intent is not to call the actual CLIMEDO API, the you can mock the `queryPatientDiary` method.

Reply:
Implemented.